dot/parachain/backing/statement_table.go

```diff
@@ -113,25 +121,246 @@ func (table *statementTable) getCommittedCandidateReceipt(candidateHash parachai
 	return data.candidate, nil
 }
 
-func (statementTable) importStatement( //nolint:unused
-	ctx *TableContext, statement parachaintypes.SignedFullStatementWithPVD,
+func (table *statementTable) importStatement(
+	tableCtx *tableContext, signedStatement parachaintypes.SignedFullStatement,
 ) (*Summary, error) {
-	// TODO: Implement this method
-	return nil, nil
+	var summary *Summary
+	var misbehaviour parachaintypes.Misbehaviour
+
+	statementVDT, err := signedStatement.Payload.Value()
+	if err != nil {
+		return nil, fmt.Errorf("getting value from statement: %w", err)
+	}
+
+	switch statementVDT := statementVDT.(type) {
+	case parachaintypes.Seconded:
+		summary, misbehaviour, err = table.importCandidate(
+			signedStatement.ValidatorIndex,
+			parachaintypes.CommittedCandidateReceipt(statementVDT),
+			signedStatement.Signature,
+			tableCtx,
+		)
+	case parachaintypes.Valid:
+		summary, misbehaviour, err = table.validityVote(
+			signedStatement.ValidatorIndex,
+			parachaintypes.CandidateHash(statementVDT),
+			validityVoteWithSign{validityVote: valid, signature: signedStatement.Signature},
+			tableCtx,
+		)
+	}
+
+	if err != nil && !errors.Is(err, errCandidateDataNotFound) {
```

Comment:
what happens if the error is `errCandidateDataNotFound`? Is the `summary` still valid?

Reply:
We always call the `postImportStatement` method after getting a summary from the `importStatement` method.
In the `postImportStatement' method, we have a logic to issue new misbehaviours.

If the summary is nil, that is misbehaviour. We issue new misbehaviour in the `postImportStatement' method to address this.

Reply:
Nope,
It is just in rust they have returned None in this case so I have returned nil.
We can return error, it will just add extra log.

Reply:
I don't understand.  You're not returning `nil` if `err != nil && error.Is(err, errCandidateDataNotFound)`, it continues with the rest of the function.

Reply:
In case of errCandidateDataNotFound error. It will return nil error.
And for other errors it will return error itself